packages/server/src/parse-chokidar-rules-from-gitignore.ts

```diff
@@ -0,0 +1,55 @@
+import parseGitignore from 'parse-gitignore'
+import fs from 'fs'
+import {resolve} from 'path'
+import _ from 'lodash'
+
+function getAllGitIgnores(rootFolder: string) {
+  function getGitIgnoresRecursively(folder: string, prefix: string): {gitIgnore: string; prefix: string}[] {
+    const entries = fs.readdirSync(folder, {withFileTypes: true})
+    const gitIgnoreFile = entries.find((dirent) => dirent.isFile() && dirent.name === '.gitignore')
+    const gitIgnore = !!gitIgnoreFile && fs.readFileSync(resolve(folder, '.gitignore'), {encoding: 'utf8'})
+
+    const subdirs = entries.filter((e) => e.isDirectory()).map((e) => e.name)
+    const gitIgnoresInSubDirs = subdirs.flatMap((subdir) =>
+      getGitIgnoresRecursively(resolve(folder, subdir), resolve(prefix, subdir)),
+    )
+
+    if (gitIgnore) {
+      return gitIgnoresInSubDirs.concat({gitIgnore, prefix})
+    } else {
+      return gitIgnoresInSubDirs
+    }
+  }
+
+  return getGitIgnoresRecursively(rootFolder, '')
+}
+
+export function parseChokidarRulesFromGitignore(rootFolder: string) {
+  const result: {ignoredPaths: string[]; includePaths: string[]} = {
+    includePaths: [],
+    ignoredPaths: [],
+  }
+
+  for (const {gitIgnore, prefix} of getAllGitIgnores(rootFolder)) {
```

Comment:
Perhaps the following block could be its own unit tested function? Just that it is is unclear on first pass the intent around the prefix stuff.